android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java

```diff
@@ -977,9 +979,20 @@ public void onPackageInstalled(List<Map<String, String>> keyboardsInstalled) {
           aPreparedCloudApiParams.add(new CloudApiTypes.CloudApiParam(
             CloudApiTypes.ApiTarget.KeyboardLexicalModels, url).setType(CloudApiTypes.JSONType.Array));
 
-          CloudDownloadMgr.getInstance().executeAsDownload(
-            context, _downloadid, null, _callback,
-            aPreparedCloudApiParams.toArray(new CloudApiTypes.CloudApiParam[0]));
+          Toast errorToast = Toast.makeText(context,
+            context.getString(com.keyman.engine.R.string.update_check_unavailable),
+            Toast.LENGTH_SHORT);
+
+          try {
+            CloudDownloadMgr.getInstance().executeAsDownload(
+              context, _downloadid, null, _callback,
+              aPreparedCloudApiParams.toArray(new CloudApiTypes.CloudApiParam[0]));
+          } catch (DownloadManagerDisabledException e) {
+            errorToast.show();
+          } catch (Exception e) {
+            errorToast.show();
+            KMLog.LogException(TAG, "Unexpected exception occurred during download attempt", e);
+          }
```

Comment:
This pattern is repeated multiple times through this changeset. Would it be better to make it into a wrapper for executeAsDownload?

Reply:
Note that `executeAsDownload` is technically accessible as an API endpoint on the engine, though undocumented.  Are we suggesting to hide the original and make the 'wrapper' the new version of the endpoint?

Reply:
Yes, I think so. The Android APIs still need deeper review, and we have toasts being raised from both the app and the engine (ideally that would not happen, it'd be up to the app to show any errors), so putting the wrapper into engine won't really make things worse.